src/main/java/com/shc/silenceengine/graphics/cameras/FPSCamera.java

```diff
@@ -139,13 +140,20 @@ public FPSCamera moveDown(float amount)
 
     public FPSCamera rotateX(float angle)
     {
+        // If the current angle is already on the limit, do nothing.
+        if (angleX <= -ANGLE_LIMIT_X || angleX >= ANGLE_LIMIT_X) {
```

Comment:
There is an issue in this line, we have to do rotation, but if the new angle exceeds the limit, we have to clamp it. This will not let the rotation happen, even if we are inside the limit.